### PR TITLE
Bootstrap changelog service

### PR DESCRIPTION
**Pull request by dolgopolovwork**
### Description of the Change
The first version of bootstrap changelog service has been created. Two endpoints are available:
1) `/changelog/execute/changelogScript `- executes changelog groovy script
2) `/changelog/execute/changelogFile` - executes changelog groovy file

All changelog transactions are created and signed by `superuser@bootstrap` account. Sample changelog script is located in `deploy/bootstrap/changelogs` folder

### CODE REVIEW

File: bootstrap/build.gradle [hunk 1]
```diff
@@ -65,11 +65,49 @@ dependencies {
     //Web3j
     compile 'org.web3j:crypto:4.1.1'
     compile 'org.web3j:core:4.1.1'
+    compile project(":notary-iroha-integration-test")
```
Comment (dolgopolovwork) [inline bootstrap/build.gradle]: We don't need this.

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/ChangelogController.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    fun executeFileChangelog(@RequestBody request: ChangelogFileRequest): ResponseEntity<Conflictable> {
+        return checkRequestAndExecute(request.details) { changelogExecutor.execute(request) }
+    }
+
+    /**
+     * Checks and executes changelog
+     * @param request - changelog request to check
+     * @param executor - execution logic
+     */
+    private fun checkRequestAndExecute(
```
Comment (dolgopolovwork) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/ChangelogController.kt]: Think about refactor

File: bootstrap/src/main/java/jp/co/soramitsu/bootstrap/changelog/ChangelogInterface.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import javax.xml.bind.DatatypeConverter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Interface of changelog scripts
+ */
+public interface ChangelogInterface {
+
+    // Name of superuser
```
Comment (yurivin) [inline bootstrap/src/main/java/jp/co/soramitsu/bootstrap/changelog/ChangelogInterface.java]: Where is method to get ChangeLog schema version ?

File: deploy/bootstrap/changelogs/sample_changelog.groovy [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import jp.co.soramitsu.bootstrap.dto.AccountPublicInfo
+import jp.co.soramitsu.bootstrap.dto.Peer
+import jp.co.soramitsu.iroha.java.Transaction
+import org.jetbrains.annotations.NotNull
+
+//Sample changelog
+class SampleChangeLog implements ChangelogInterface {
+
+    @Override
+    List<Transaction> createChangelog(@NotNull List<AccountPublicInfo> accounts,
```
Comment (dolgopolovwork) [inline deploy/bootstrap/changelogs/sample_changelog.groovy]: Make it return one transaction

**Comment by Mingela** [comment]
lgtm

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/IrohaAPIExt.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        .onTransactionCommitted { successTx -> txStatus.txHash = successTx.txHash.toUpperCase() }
+        .onMstPending { pendingTx -> txStatus.txHash = pendingTx.txHash.toUpperCase() }
+        .build()
+}
+
+/**
+ * Send list of transactions to Iroha as BATCH and check if it is committed
+ * @param lst - list of built protobuf iroha transactions
+ */
```
Comment (yurivin) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/IrohaAPIExt.kt]: We had a conversation to create function which will not wait untill transaction will be commited and return in pending state.

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package jp.co.soramitsu.bootstrap.iroha
+
+import jp.co.soramitsu.bootstrap.dto.IrohaConfig
+import jp.co.soramitsu.iroha.java.IrohaAPI
+import org.springframework.stereotype.Component
+import java.io.Closeable
+
+/**
+ * Lazy Iroha API pool
+ */
```
Comment (yurivin) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt]: This pool looks like a Dummy pool. It should have ability to check state of connection and remove connection if expired. This realisation may cause memory leaks. So should be upgraded in the future.

Comment (dolgopolovwork) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt]: Please take a look. It's fixed now.

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt [hunk 7]
```diff
@@ -13,6 +16,15 @@ class LazyIrohaAPIPool : Closeable {
 
     // Collection of IrohaAPIs
     private val apiPool = HashMap<IrohaConfig, IrohaAPI>()
+    // Executor service that manages old and stale IrohaAPIs
+    private val scavengerExecutorService = Executors.newSingleThreadScheduledExecutor(
+        ThreadFactoryBuilder().setNameFormat("iroha-api-scavenger").build()
+    )
+
+    init {
+        // Removes old APIs every 5 minutes
```
Comment (Mingela) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt]: What if the connection dies in < 5 minutes? Or it does not matter from `IrohaAPI` perspective?

Comment (dolgopolovwork) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/iroha/LazyIrohaAPIPool.kt]: Nothing bad will happen, probably. If you accidentally took stale IrohaAPI, new API will be created.

**Pull request merged**